File: js-jest/src/item_updaters/aged_brie_updater.js
const { ItemUpdater } = require("../gilded_rose");

class AgedBrieUpdater extends ItemUpdater {
    updateQuality() {
        
        //The Quality of an item is never more than 50
        if(this.item.quality > 50){
            this.item.quality= 50;
        }

        if (this.item.quality > 0 && this.item.quality < 50) {
            this.item.quality = this.item.quality + 1;
        }

        return this.item;
    }
}

module.exports = {
    AgedBrieUpdater
}

File: Java/src/main/java/com/gildedrose/updaters/AgedBrieUpdater.java
package com.gildedrose.updaters;

import com.gildedrose.Item;

public class AgedBrieUpdater extends ItemUpdater {
    public AgedBrieUpdater(Item item) {
        super(item);
    }

    @Override
    public void update() {
        increaseQuality(1);
        decreaseSellIn();
        if (item.sellIn < 0) {
            increaseQuality(1);
        }
    }
}
